Slim/Player/StreamingController.pm

```diff
@@ -987,9 +992,17 @@ sub _FlushGetNext {			# flush -> Idle; IF [moreTracks] THEN getNextTrack -> Trac
 	_getNextTrack($self, $params, 1);
 }
 
-sub _NextIfMore {			# -> Idle; IF [moreTracks] THEN getNextTrack -> TrackWait ENDIF
+sub _NextIfMore {			# -> Idle; IF [moreTracks] AND [roomInQueue] THEN getNextTrack -> TrackWait ENDIF
 	my ($self, $event, $params) = @_;
 	_setStreamingState($self, IDLE);
+	# if we already have a playing and a streaming track, then wait until the streaming 
+	# one starts to play before getting the next one. That happens when player's buffer 
+	# is much larger than track. 
+	if (scalar @{$self->{'songqueue'}} < 2) {
+		_getNextTrack($self, $params, 1);
+	} else {
+		$log->info("streaming track not started yet, will wait until then to try next track");
+	}
 	_getNextTrack($self, $params, 1);
```

Comment:
@philippe44 Investigating #1214 I stumbled across this change: this would call `_getNextTrack()` twice. Is this expected? Or should the old (second) call be removed? If I disable line 1002 I wouldn't get the infinite "bad state" message locking up LMS.